libs/client-sdk/src/CMakeLists.txt

```diff
@@ -46,22 +45,29 @@ target_link_libraries(${LIBNAME}
         PRIVATE Boost::headers Boost::json Boost::url launchdarkly::sse launchdarkly::internal foxy)
 
```

Comment:
The changes in most of these sub-lib `CMakeLists.txt` are:
1. Switch from `target_include_directories(${LIBNAME} PUBLIC ../include` to one that uses generator expressions, so it works at build and installation time
2. Swap out raw `lib` or `bin` paths with `GNUInstallDir` variables
3. Install the targets to our `LD_TARGETS_EXPORT_NAME`